packages/product-json-to-csv/src/map-product-data.js

```diff
@@ -39,18 +42,47 @@ export default class ProductMapping {
     )
     const mappedProduct = varWithProductInfo.map(
       (variant: SingleVarPerProduct): MappedProduct =>
-        this._mapProperties(variant)
+        ProductMapping._postProcessProduct(this._mapProduct(variant))
     )
-    const flatProducts = mappedProduct.map(flatten)
-    return flatProducts
+
+    return mappedProduct
+  }
+
+  static _postProcessProduct(originalProduct: MappedProduct): MappedProduct {
+    // move variant prices and attributes to top level
+    // variant.prices => prices
+    // variant.attributes.productSize => productSize
+    const prices: Array<Price> = _.get(
+      originalProduct,
+      'variant.prices',
+      undefined
+    )
+    const attributes: Object = _.get(originalProduct, 'variant.attributes', {})
+    const cleanedProduct = { ...originalProduct, prices, ...attributes }
+
+    delete cleanedProduct.variant.prices
+    delete cleanedProduct.variant.attributes
+
+    const product: MappedProduct = flatten(cleanedProduct)
+
+    // remove undefined values
+    Object.entries(product).forEach(([key, val]) => {
+      if (_.isUndefined(val)) delete product[key]
+    })
+
+    return product
   }
 
   // merge all variants into a `variant` property and remove
   // `masterVariant` and `variants` fields from product
   static _mergeVariants(
     product: ResolvedProductProjection
   ): ProdWithMergedVariants {
-    const variant = [product.masterVariant, ...product.variants]
+    const variant = _([product.masterVariant])
+      .concat(product.variants)
+      .compact()
+      .value()
```

Comment:
This will work only if `variants` is not `null` or `undefined`. I am not sure if this can happen.
Updated here https://github.com/commercetools/nodejs/pull/680/commits/4e06beed22d16ca96402a4d8148c25e1b0d702cb#diff-9435e999bde99eccdb8481181b8252b8R81